src/ALCProxy.Communication/CommObjects.cs

```diff
@@ -0,0 +1,65 @@
+﻿// Licensed to the .NET Foundation under one or more agreements. 
+// The .NET Foundation licenses this file to you under the MIT license. 
+// See the LICENSE file in the project root for more information. 
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.IO;
+using System.Runtime.Serialization;
+
+namespace ALCProxy.Communication
+{
+    public class ClientDispatch : ALCClient
+    {
+        public ClientDispatch(Type intType) : base(intType, "ServerDispatch`1") { }
+        protected override object SerializeParameter(object param, Type paramType)
+        {
+            MemoryStream stream = new MemoryStream();
+            //Serialize the Record object to a memory stream using DataContractSerializer.  
+            DataContractSerializer serializer = new DataContractSerializer(paramType);
+            serializer.WriteObject(stream, param);
+            return stream;
+
+        }
+        protected override object DecryptReturnType(object s, Type returnType)
+        {
+            MemoryStream stream = (MemoryStream)s;
+            stream.Position = 0;
+            //Deserialize the Record object back into a new record object.  
+            DataContractSerializer newSerializer = new DataContractSerializer(returnType);
+            return newSerializer.ReadObject(stream);
+        }
+    }
+    public class ServerDispatch<InterfaceType> : ALCServer<InterfaceType>
```

Comment:
Try to be consistent in `I`/`InterfaceType` between different classes - I think generally we prefer shorter names for type parameters when it's relatively clear from context what they are.